include/deal.II/base/index_set.h

```diff
@@ -256,6 +256,14 @@ class IndexSet
   size_type index_within_set (const size_type global_index) const;
 
   /**
+   * Same as above, but returns <code>true</code> if the global index is
+   * within the set and <code>false</code> otherwise. In the former case
+   * the how-manyth element of this set is set in @p local index.
+   */
+  bool index_within_set (const size_type global_index,
+                         size_type &local_index) const;
```

Comment:
Alternatively, because it *also* returns an integer: One could either return an integer that happens to be `numbers::unsigned_int` if the index is not in the set, or a regular `size_type` otherwise. You could implement this by just changing the implementation of the previous function (and the documentation about the assertion).

Another alternative would be to return a `boost::optional<size_type` that could be set, but doesn't have to be. That would also just extend the existing function. I don't recall if a `boost::optional<T>` is automatically convertible to a `T`, which would make it backward compatible.